docs/core/deploying/native-aot/interop.md

```diff
@@ -79,4 +79,5 @@ Examples:
 
 The Native AOT compiler exports methods annotated with <xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute> with a nonempty `EntryPoint` property as
 public C entry points. This makes it possible to either dynamically or statically link the AOT compiled modules into external
-programs. For more information, see [NativeLibrary sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary/README.md).
+programs. Only methods marked `UnmanagedCallersOnly` in the published assembly are considered: methods in project references or NuGet packages will not be exported.
+For more information, see [NativeLibrary sample](https://github.com/dotnet/samples/tree/main/core/nativeaot/NativeLibrary/README.md).
```

Comment:
Should we also include information about using `UnmanagedEntryPointsAssembly` ItemGroup for exports from referenced assemblies?

Reply:
@MichalStrehovsky ping

Reply:
We've not been documenting the internals of Xamarin bindings. I think there was an issue in the runtime repo on whether we want to do it and it was won't fixed.

Reply:
I was referring to: https://github.com/dotnet/runtime/pull/83396 where we were introducing the `ItemGroup` for a more general use case (not just Xamarin support).

Reply:
Yep, and it was introduced for Xamarin. I don't know if this is ready for general consumption. For example we don't handle the situation when two UnmanagedCallersOnly methods have the same EntryPoint. This results in a bad crash. Allowing NuGets to add themselves to `UnmanagedEntryPointsAssembly` makes it more likely people are going to hit the bad failure modes.

We can easily detect conflicting entrypoints _within_ a single assembly with an analyzer (we don't have such analyzer, but it wouldn't be hard to make ont). We cannot detect a conflict when multiple assemblies are involved until it's too late. We don't have any recourse than to just fail the AOT compilation.